netmaster/master/network.go

```diff
@@ -162,34 +163,55 @@ func CreateNetwork(network intent.ConfigNetwork, stateDriver core.StateDriver, t
 }
 
 func attachServiceContainer(tenantName, networkName string, stateDriver core.StateDriver) error {
-	contName := tenantName + "dns"
+	contName := getDNSName(tenantName)
 	docker, err := utils.GetDockerClient()
 	if err != nil {
 		log.Errorf("Unable to connect to docker. Error %v", err)
 		return err
 	}
 
+	cinfo, err := docker.InspectContainer(contName)
+	if err != nil {
+		if strings.Contains(err.Error(), "no such id") {
+			// DNS container not started for this tenant. Start skydns container
+			err = startServiceContainer(tenantName)
+			if err != nil {
+				log.Warnf("Error starting service container. "+
+					"Continuing without DNS provider. Error: %v", err)
+				return nil
+			}
+		}
+	}
+
+	// If it's not in running state, restart the container.
+	// This case can occur if the host is reloaded
+	if !cinfo.State.Running {
+		log.Debugf("Container %s not running. Restarting the conttainer", contName)
+		err = docker.RestartContainer(contName, 0)
```

Comment:
You might want to test this scenario. I'm not sure if RestartContainer is the right call to make.
If container is in stopped state, you might need to do just StartContainer()